pkg/controller/accountclaim/organizational_units.go

```diff
@@ -0,0 +1,209 @@
+	})
+
+	// Search for ConfigMap that holds OU mapping
+	instance := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: awsv1alpha1.AccountCrNamespace, Name: OUConfigMapName}, instance)
+	if err != nil {
+		// If we failed to retrieve the ConfigMap, simply leave the account in Root
+		unexpectedErrorMsg := fmt.Sprintf("OU: Failed to find OU mapping ConfigMap, leaving account in root")
+		reqLogger.Info(unexpectedErrorMsg)
+		accountClaim.Spec.AccountOU = "ROOT"
+		return r.specUpdate(reqLogger, accountClaim)
+	}
+
+	// Get OU ID for root and base
+	friendlyOUName := account.Spec.LegalEntity.ID
+	baseID, rootID, err := checkOUMapping(instance)
+	if err != nil {
+		invalidOUErrorMsg := fmt.Sprintf("Invalid OU ConfigMap, missing root and/or base fields: %s", instance.Data)
+		reqLogger.Error(err, invalidOUErrorMsg)
+		return err
+	}
+
+	// Log ConfigMap
+	configmapMsg := fmt.Sprintf("OU: OU mapping ConfigMap was found in namespace: %s, root: %s, base: %s", OUConfigMapName, rootID, baseID)
+	reqLogger.Info(configmapMsg)
+
+	// Create/Find account OU
+	OUID, err := CreateorFindOU(reqLogger, awsClient, account, friendlyOUName, baseID)
+	if err != nil {
+		return err
+	}
+	err = MoveAccount(reqLogger, awsClient, account, OUID, rootID)
+	if err != nil {
+		// If error was cause by the account already being inside the OU, simply update the accountclaim cr and returns
+		if err == awsv1alpha1.ErrAccAlreadyInOU {
+			// Log account already in desired location
+			accountMovedMsg := fmt.Sprintf("OU: Account %s was already in the desired OU %s", account.Name, account.Spec.LegalEntity.ID)
+			reqLogger.Info(accountMovedMsg)
+			// Update accountclaim spec
+			accountClaim.Spec.AccountOU = OUID
+			return r.specUpdate(reqLogger, accountClaim)
+		}
+		return err
+	}
+
+	// Log account moved successfully
+	accountMovedMsg := fmt.Sprintf("OU: Account %s successfully moved to OU %s", account.Name, account.Spec.LegalEntity.ID)
+	reqLogger.Info(accountMovedMsg)
+
+	// Update unclaimedAccount.Spec.AwsAccountOU
+	accountClaim.Spec.AccountOU = OUID
+	return r.specUpdate(reqLogger, accountClaim)
+}
+
+// CreateorFindOU will create or find an existing OU and return its ID
+func CreateorFindOU(reqLogger logr.Logger, client awsclient.Client, account *awsv1alpha1.Account, friendlyOUName string, baseID string) (string, error) {
+	// Create/Find account OU
+	createCreateOrganizationalUnitInput := organizations.CreateOrganizationalUnitInput{
+		Name:     &friendlyOUName,
+		ParentId: &baseID,
+	}
+	ouOutput, ouErr := client.CreateOrganizationalUnit(&createCreateOrganizationalUnitInput)
+	if ouErr != nil {
+		if aerr, ok := ouErr.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "DuplicateOrganizationalUnitException":
+				duplicateOUMsg := fmt.Sprintf("OU: %s Already exists", account.Spec.LegalEntity.ID)
+				reqLogger.Info(duplicateOUMsg)
+				return findOUIDFromName(reqLogger, client, baseID, friendlyOUName)
```

Comment:
What happens if this returns an error? Does it need any special handling?

Reply:
Possible errors are already handled

Reply:
Where?

Reply:
In the [method](https://github.com/openshift/aws-account-operator/blob/7f5d2f824385e8c296998fca99ee1577c829bb1e/pkg/controller/accountclaim/organizational_units.go#L170) itself I log unexpected errors and I handle the return [values] when I get them(https://github.com/openshift/aws-account-operator/blob/7f5d2f824385e8c296998fca99ee1577c829bb1e/pkg/controller/accountclaim/organizational_units.go#L52-L55)

Reply:
In `findOUIDFromName` you only handle if there's an aws error, not if there's some generic error. There's also no specific handling for any aws error, so why unwrap it if you're going to return it straight up to the parent function? Line 176 of this file. I'm wondering if you're going to specifically handle the error from findOUIDFromName anywhere.

Reply:
I see what you’re saying about making sure I catch generic errors as well, I’ll fix that. I always unwrap AWS errors because I was told it is good practice + it helps with debugging if there is ever a need but I am open to other opinions.

Reply:
You should unwrap it. I just want to verify that there's nothing specific you want to do with that error based on its message.

Reply:
I see, well unlike some other parts of this PR where I use specific errors as a queue to apply different logic this particular case any error would be bad. That's why I just log and return.